### ISSUE TITLE
[idea] Use Xamarin ImageSource to set the Icon

### ISSUE THREAD

**Issue by galadril** [Mapsui/Mapsui#716]
I think that it would be an improvement if you can set the Pin Icon from a ImageSource. (Xamarin.Forms.ImageSource).

What do you think?

**Comment by pauldendulk** [comment]
Well I would get most enthusiastic about a solution that has no dependency on Xamarin.Forms. 

How would that work? Why is it better? (note, I am only partially involved in the Xamarin.Forms part).

**Comment by galadril** [comment]
True, but images are something really specific on mobile devices. If you use ImageSource, Xamarin Forms takes the correct image for your screen size. So it takes for image from the correct drawable directory in Android:
![image](https://user-images.githubusercontent.com/14561640/66804361-8a50fa00-ef22-11e9-86ad-c5339f9d182a.png)

and in iOS
![image](https://user-images.githubusercontent.com/14561640/66804393-9dfc6080-ef22-11e9-867f-798ffdd213ce.png)


For example, now i can only have an marker icon image in the Forms shared project, as an embedded resource.. thats one image with one image size.. thats going to look different on a smaller device like iPhone SE, or on an iPhone X. You need different images per pdi/screen density

**Comment by galadril** [comment]
Since Xamarin Forms 4.0 they really moved all the image handling to ImageSources:
https://devblogs.microsoft.com/xamarin/introducing-xamarin-forms-4-0-the-era-of-shell/

Its also possible to implement your own Custom ImageSource ofcourse
https://dev.to/muak_x/creating-custom-imagesource-in-xamarinforms-and-displaying-an-svg-image-to-default-image-controll-4dn7

**Comment by pauldendulk** [comment]
Ah, I didn't think about the image resolution issue. Something to keep in mind for our design. Btw, there is also support for svg in Mapsui.

But for the maintainability of the project, at least with respect to my own involvement, I am move more functionality away from XF. It should be possible to add custom platform functionality like this though.

**Comment by charlenni** [comment]
@galadril That's a good idea. I happily awaiting your PR.
